skins/cat17/address-form/src/store.ts

```diff
@@ -0,0 +1,138 @@
+                value: '',
+                isValid: Validity.INCOMPLETE
+            },
+            lastName: {
+                dataEntered: false,
+                value: '',
+                isValid: Validity.INCOMPLETE
+            },
+            companyName: {
+                dataEntered: false,
+                value: '',
+                isValid: Validity.INCOMPLETE
+            },
+            street: {
+                dataEntered: false,
+                value: '',
+                isValid: Validity.INCOMPLETE
+            },
+            postcode: {
+                dataEntered: false,
+                value: '',
+                isValid: Validity.INCOMPLETE
+            },
+            city: {
+                dataEntered: false,
+                value: '',
+                isValid: Validity.INCOMPLETE
+            },
+            country: {
+                dataEntered: false,
+                value: '',
+                isValid: Validity.INCOMPLETE
+            }
+        }  
+	},
+	mutations: {
+        [STORE_VALUES](state, payload): void {
+            payload.forEach((field: FormData) => {
+               state.form[field.name].value = field.value;
+           });
+        },
+        [VALIDATE_INPUT](state, payload): void {
+            payload.forEach((field: FormData) => {
+                if (field.value === '' && field.optionalField === true) {
+                    state.form[field.name] = {
+                        ...state.form[field.name],
+                        dataEntered: false,
+                        isValid: Validity.INCOMPLETE
+                    };
+                }
+                else {
+                    state.form[field.name] = {
+                        ...state.form[field.name],
+                        dataEntered: true,
+                        isValid: Helper.inputIsValid(field.value, field.pattern)
+                    };
+                }
+            });
+		},
+		[MARK_EMPTY_FIELD_INVALID](state, payload): void {
+			payload.forEach((field: FormData) => {
+                if (!field.optionalField && state.form[field.name].isValid === Validity.INCOMPLETE) {
+                    state.form[field.name].isValid = Validity.INVALID;
+				}
+			});
+		},
+		[BEGIN_ADDRESS_VALIDATION](state): void {
+			state.isValidating = true;
+		},
+        [FINISH_ADDRESS_VALIDATION](state, payload: ValidationResult): void {
+			ADDRESS_FIELD_VALIDATOR_NAMES.forEach(name => {
+                if (state.form[name].dataEntered === true) {
+                    state.form[name] = {
+                        ...state.form[name],
+                        isValid: state.form[name].isValid !== false && !payload.messages[name] ? Validity.VALID : Validity.INVALID
+					};
+				}
+			});
+			state.isValidating = false;
+		}
+	},
+	getters: {
+		validity(state, contentName: string) {
+            return state.form[contentName].isValid;
+		}
+	},
+	actions: {
+        storeAddressFields({ commit }, payload: Payload) {
```

Comment:
I imagined this more as `storeAddressField` with the payload of a single field. Is there any advantage to always sending all the field values from vue when one field value changes?